### ISSUE TITLE
Docker build tasks are failing since today: invalid argument --tag

### ISSUE THREAD

**Issue by brainbug89** [microsoft/azure-pipelines-tasks#14298]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: Docker build  

## Environment
Azure Pipelines
Hosted Agent

## Issue Description

Since today all Docker build tasks are failing in our pipeline:
```
Starting: Build Backend Container
==============================================================================
Task         : Docker
Description  : Build or push Docker images, login or logout, start or stop containers, or run a Docker command
Version      : 2.181.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-docker-tsg
==============================================================================
/usr/bin/docker build -f /home/vsts/work/1/s/backend/Dockerfile --label com.visualstudio.XXXXXXXX.image.system.teamfoundationcollectionuri=https://XXXXXXXX.visualstudio.com/ --label com.visualstudio.XXXXXXXX.image.system.teamproject=XXXXXXXX --label com.visualstudio.XXXXXXXX.image.build.repository.name=prj-foobar --label com.visualstudio.XXXXXXXX.image.build.sourceversion=64c164c6cca0a26aaa42770e41d4877dedbf4363 --label com.visualstudio.XXXXXXXX.image.build.repository.uri=
invalid argument "***/foobar/foobar-backend:" for "-t, --tag" flag: invalid reference format
See 'docker build --help'.
##[error]invalid argument "***/foobar/foobar-backend:" for "-t, --tag" flag: invalid reference format
##[error]See 'docker build --help'.
##[error]The process '/usr/bin/docker' failed with exit code 125
Finishing: Build Backend Container
```

The same pipeline (no changes!) worked fine yesterday:
```
Starting: Build Backend Container
==============================================================================
Task         : Docker
Description  : Build or push Docker images, login or logout, start or stop containers, or run a Docker command
Version      : 2.176.0
Author       : Microsoft Corporation
Help         : https://aka.ms/azpipes-docker-tsg
==============================================================================
/usr/bin/docker build -f /home/vsts/work/1/s/backend/Dockerfile --label com.visualstudio.XXXXXXXX.image.system.teamfoundationcollectionuri=https://XXXXXXXX.visualstudio.com/ --label com.visualstudio.XXXXXXXX.image.system.teamproject=XXXXXXXX --label com.visualstudio.XXXXXXXX.image.build.repository.name=prj-foobar --label com.visualstudio.XXXXXXXX.image.build.sourceversion=5ee2b484bf8283433a5e99cab949a2bcd0452b98 --label com.visualstudio.XXXXXXXX.image.build.repository.uri=
Sending build context to Docker daemon  138.3MB

Step 1/26 : [...cut...]
```
The only difference is the task version.

Here is the task definition:
```
    - task: Docker@2
      displayName: Build Backend Container
      inputs:
        containerRegistry: $(dockerRegistryServiceConnection)
        repository: '$(namespace)/$(image_name_backend)'
        command: 'build'
        Dockerfile: 'backend/Dockerfile'
        buildContext: '.'
        tags: $(image_tags)
        arguments: '--build-arg ENV_FOOBAR_GENERAL_BUILDVERSION=$(Build.BuildNumber)'
```

The images tags are defined like this:
```
  image_tags: |
    $(Build.SourceBranchName)
    $(Build.BuildId)
    $(Build.SourceVersion)
```

**Comment by shigupt202** [comment]
Please refer this [issue](https://github.com/microsoft/azure-pipelines-tasks/issues/14264). We have made a fix from our end, it should reach out to all the customers in few days. But in the mean time you can try the workarounds mentioned in the issue.

**Comment by brainbug89** [comment]
Workaround is working. Thanks!

```
  image_tags: |-
    $(Build.SourceBranchName)
    $(Build.BuildId)
    $(Build.SourceVersion)
```

**Comment by brainbug89** [comment]
Just saw that the task now uses the fixed version (2.181.2). So the workaround is probably no longer necessary.
